test/gtest/uct/ib/test_ib_pkey.cc

```diff
@@ -214,6 +214,48 @@ UCS_TEST_P(test_uct_ib_pkey, test_pkey_pairs) {
                                                  (uct_device_addr_t*)ib_addr1,
                                                  NULL));
 
+        const uct_iface_is_reachable_params_t params1      = {
+            .field_mask         = 0,
+            .device_addr        = (uct_device_addr_t*)ib_addr1,
+            .iface_addr         = NULL,
+            .info_string        = (char*)ucs_malloc(4096, "test ib pkey"),
+            .info_string_length = 4096
+        };
+
+        const uct_iface_is_reachable_params_t params2      = {
+            .field_mask         = 0,
+            .device_addr        = (uct_device_addr_t*)ib_addr2,
+            .iface_addr         = NULL,
+            .info_string        = (char*)ucs_malloc(4096, "test ib pkey"),
+            .info_string_length = 4096
+        };
+
+        const uct_iface_is_reachable_params_t params_null1 = {
+            .field_mask         = 0,
+            .device_addr        = (uct_device_addr_t*)ib_addr1,
+            .iface_addr         = NULL,
+            .info_string        = NULL,
+            .info_string_length = 0
+        };
+
+        const uct_iface_is_reachable_params_t params_null2 = {
+            .field_mask         = 0,
+            .device_addr        = (uct_device_addr_t*)ib_addr2,
+            .iface_addr         = NULL,
+            .info_string        = NULL,
+            .info_string_length = 0
+        };
+
+        EXPECT_EQ(res, uct_ib_iface_is_reachable_v2(m_e1->iface(), &params1));
```

Comment:
You check that the iface can reach itself? Why check it twice?

Reply:
It's just for testing v2 functions. One is for non-null buffer another is for null buffer. Thanks

Reply:
Null buffer testcase has been removed since the v2 function only allows non-null buffer parameter. Actually, it wouldn't reach here when buffer is null because uct_iface_is_reachable_v2 will check the filed mask to make sure the inputting parameters is available. Thanks.

Reply:
Add test null buffer scenario:

```

        /* Test reachable v2 API without string buffer */
        const uct_iface_is_reachable_params_t params_null1      = {
            .field_mask         = UCT_IFACE_IS_REACHABLE_FIELD_DEVICE_ADDR |
                                  UCT_IFACE_IS_REACHABLE_FIELD_IFACE_ADDR,
            .device_addr        = (uct_device_addr_t*)ib_addr1,
            .iface_addr         = NULL
        };

        const uct_iface_is_reachable_params_t params_null2      = {
            .field_mask         = UCT_IFACE_IS_REACHABLE_FIELD_DEVICE_ADDR |
                                  UCT_IFACE_IS_REACHABLE_FIELD_IFACE_ADDR,
            .device_addr        = (uct_device_addr_t*)ib_addr2,
            .iface_addr         = NULL
        };

        EXPECT_EQ(res, uct_ib_iface_is_reachable_v2(m_e1->iface(), &params_null1));
        EXPECT_EQ(res, uct_ib_iface_is_reachable_v2(m_e2->iface(), &params_null2));
```